From 2339a9e09b60a51de81cb6a4e703d12d8d38874a Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 30 Jul 2018 10:58:20 +0800
Subject: [PATCH 4289/5242] MLK-19088-1 ARM64: imx: change the clkreq to
 opendrain input

commit  52b250d04d06e594712f7c089db9e4df0574c65c from
https://source.codeaurora.org/external/imx/linux-imx.git

In the L1.1 ASPM implementation, the CLK_REQ# should be
configured as open drain, pull up and input mode.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |    6 ++----
 include/dt-bindings/pinctrl/pins-imx8mq.h        |    4 ++--
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index e01718d..c820557 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -204,7 +204,7 @@
 
 		pinctrl_pcie0: pcie0grp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_I2C4_SCL_GPIO5_IO20	0x16
+				MX8MQ_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B	0x76 /* open drain, pull up */
 				MX8MQ_IOMUXC_UART4_TXD_GPIO5_IO29	0x16
 				MX8MQ_IOMUXC_UART4_RXD_GPIO5_IO28	0x16
 			>;
@@ -212,7 +212,7 @@
 
 		pinctrl_pcie1: pcie1grp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_I2C4_SDA_GPIO5_IO21	0x16
+				MX8MQ_IOMUXC_I2C4_SDA_PCIE2_CLKREQ_B	0x76 /* open drain, pull up */
 				MX8MQ_IOMUXC_ECSPI2_SCLK_GPIO5_IO10	0x16
 				MX8MQ_IOMUXC_ECSPI2_MISO_GPIO5_IO12	0x16
 			>;
@@ -658,7 +658,6 @@
 &pcie0{
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie0>;
-	clkreq-gpio = <&gpio5 20 GPIO_ACTIVE_LOW>;
 	disable-gpio = <&gpio5 29 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio5 28 GPIO_ACTIVE_LOW>;
 	ext_osc = <1>;
@@ -669,7 +668,6 @@
 &pcie1{
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie1>;
-	clkreq-gpio = <&gpio5 21 GPIO_ACTIVE_LOW>;
 	disable-gpio = <&gpio5 10 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio5 12 GPIO_ACTIVE_LOW>;
 	ext_osc = <1>;
diff --git a/include/dt-bindings/pinctrl/pins-imx8mq.h b/include/dt-bindings/pinctrl/pins-imx8mq.h
index 0e1d67d..498b76f 100644
--- a/include/dt-bindings/pinctrl/pins-imx8mq.h
+++ b/include/dt-bindings/pinctrl/pins-imx8mq.h
@@ -564,12 +564,12 @@
 #define MX8MQ_IOMUXC_I2C3_SDA_TPSMP_HDATA21                                 0x228 0x490 0x000 0x7 0x0
 #define MX8MQ_IOMUXC_I2C4_SCL_I2C4_SCL                                      0x22C 0x494 0x000 0x0 0x0
 #define MX8MQ_IOMUXC_I2C4_SCL_PWM2_OUT                                      0x22C 0x494 0x000 0x1 0x0
-#define MX8MQ_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B                                0x22C 0x494 0x524 0x2 0x0
+#define MX8MQ_IOMUXC_I2C4_SCL_PCIE1_CLKREQ_B                                0x22C 0x494 0x524 0x12 0x0
 #define MX8MQ_IOMUXC_I2C4_SCL_GPIO5_IO20                                    0x22C 0x494 0x000 0x5 0x0
 #define MX8MQ_IOMUXC_I2C4_SCL_TPSMP_HDATA22                                 0x22C 0x494 0x000 0x7 0x0
 #define MX8MQ_IOMUXC_I2C4_SDA_I2C4_SDA                                      0x230 0x498 0x000 0x0 0x0
 #define MX8MQ_IOMUXC_I2C4_SDA_PWM1_OUT                                      0x230 0x498 0x000 0x1 0x0
-#define MX8MQ_IOMUXC_I2C4_SDA_PCIE2_CLKREQ_B                                0x230 0x498 0x528 0x2 0x0
+#define MX8MQ_IOMUXC_I2C4_SDA_PCIE2_CLKREQ_B                                0x230 0x498 0x528 0x12 0x0
 #define MX8MQ_IOMUXC_I2C4_SDA_GPIO5_IO21                                    0x230 0x498 0x000 0x5 0x0
 #define MX8MQ_IOMUXC_I2C4_SDA_TPSMP_HDATA23                                 0x230 0x498 0x000 0x7 0x0
 #define MX8MQ_IOMUXC_UART1_RXD_UART1_DCE_RX                                 0x234 0x49C 0x4F4 0x0 0x0
-- 
1.7.9.5

